app/js/components/homePage/manageApps/manageApps.jsx

```diff
@@ -54,22 +58,80 @@ export default class ManageApps extends React.Component {
     const url = location.href.split('/')[3];
     let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
     this.requestUrl ='/owa/addapp';
+    let that = this;
 
     const addonFile = new FormData();
     addonFile.append('file', document.getElementById('fileInput').files[0]);
-    $.ajax({
+  
+    const response = $.ajax({
+      xhr: function(){
+        let xhrRequest = $.ajaxSetup().xhr();
+        if(xhrRequest.upload){
+          xhrRequest.upload.addEventListener('progress',that.handleProgress, false);
+        }
+        return xhrRequest;
+       
+      },
       type: "POST",
       url: `https:/${apiBaseUrl}${this.requestUrl}`,
       data: addonFile,
       contentType: false,
       processData: false,
       cache: false,
       success: function (result) {
+        this.setState({
+          msgBody: "App has been successfully installed",
+          msgType: "success",
+          showMsg: true,
+          appList: result,
+          staticAppList: result
+        });
+
+        if(that.state.showMsg === true) {
+          setTimeout(that.handleAlertBehaviour, 2000);
```

Comment:
Hi @paulupendo, please why are you setting timeout here?

Reply:
I am using the timeout to auto hide the alert messages after they show up. Time is set to 2 seconds. Similarly to how a toastr notification would behave.